src/read/pe/pe_export.rs

```diff
@@ -0,0 +1,97 @@
+use core::fmt::Debug;
+
+/// Possible exports from a PE file
+#[derive(Clone)]
+pub enum PeExport<'data> {
+    /// A named exported symbol from this PE file
+    Regular {
+        /// The ordinal of this export
+        ordinal: u32,
+        /// The name of this export
+        name: &'data [u8],
+        /// The virtual address pointed to by this export
+        address: u64,
+    },
+
+    /// An export that only has an ordinal, but no name
+    ByOrdinal {
+        /// The ordinal of this export
+        ordinal: u32,
+        /// The virtual address pointed to by this export
+        address: u64,
+    },
+
+    /// A forwarded export (i.e. a symbol that is contained in some other DLL).
+    /// This concept is [PE-specific](https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#export-address-table)
+    Forwarded {
+        /// The ordinal of this export
+        ordinal: u32,
+        /// The name of this export
+        name: &'data [u8],
+        /// The name of the actual symbol, in some other lib.
+        /// for example, "MYDLL.expfunc" or "MYDLL.#27"
+        forwarded_to: &'data [u8],
+    },
+
+    /// A forwarded export (i.e. a symbol that is contained in some other DLL) that has no name.
+    /// This concept is [PE-specific](https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#export-address-table)
+    ForwardedByOrdinal {
+        /// The ordinal of this export
+        ordinal: u32,
+        /// The name of the actual symbol, in some other lib.
+        /// for example, "MYDLL.expfunc" or "MYDLL.#27"
+        forwarded_to: &'data [u8],
+    },
+}
+
+impl<'a> Debug for PeExport<'a> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::result::Result<(), core::fmt::Error> {
+        match &self {
+            PeExport::Regular {
+                ordinal,
+                name,
+                address,
+            } => f
+                .debug_struct("Regular")
+                .field("ordinal", &ordinal)
+                .field(
+                    "data",
+                    &core::str::from_utf8(name).unwrap_or("<invalid name>"),
```

Comment:
Should this print the raw bytes when the name is not valid utf8?

Reply:
Done. But that's kind of ugly code...

Reply:
Just a suggestion: you could make it closer to what you had before by fixing the `Debug` impl for `ByteString` to have the output you want (because it probably should anyway), and then do `.field("name", &ByteString(name))`. `ByteString` exists purely for a nicer debug output.

Reply:
Hmmm. I had forgotten about `ByteString` because I misused it at first (trying to use this private type into public interfaces) and wasn't willing to use it anymore...but in this `Debug` impl that's definitely what I need! Thanks for the tip.

> you could make it closer to what you had before by fixing the Debug impl for ByteString

Hmm. I would say that the current use of `String::from_utf8_lossy` is fine. I think that, in case of invalid UTF-8, most users would prefer having some UTF replacement characters than a list of somewhat garbled bytes, since that's just for debug purposes (at least I would :-) )